SYCL/ESIMD/api/functional/operators/operator_bitwise_not.hpp

```diff
@@ -0,0 +1,171 @@
+
+    mutate(ref_data, For_bitwise_not<DataT>());
+
+    // If current number of elements is equal to one, then run test with each
+    // one value from reference data.
+    // If current number of elements is greater than one, then run tests with
+    // whole reference data.
+    if constexpr (NumElems == 1) {
+      for (size_t i = 0; i < ref_data.size(); ++i) {
+        passed = run_verification(queue, {ref_data[i]}, data_type);
+      }
+    } else {
+      passed = run_verification(queue, ref_data, data_type);
+    }
+    return passed;
+  }
+
+private:
+  bool run_verification(sycl::queue &queue, const std::vector<DataT> &ref_data,
+                        const std::string &data_type) {
+    assert(ref_data.size() == NumElems &&
+           "Reference data size is not equal to the simd vector length.");
+
+    bool passed = true;
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         allocator);
+    shared_vector<DataT> source_simd_result(NumElems, allocator);
+    shared_vector<DataT> operator_result(NumElems, allocator);
+
+    shared_element<bool> is_correct_type(queue, true);
+
+    queue.submit([&](sycl::handler &cgh) {
+      const DataT *const ref = shared_ref_data.data();
+      DataT *const source_simd_result_data_ptr = source_simd_result.data();
+      DataT *const operator_result_data_ptr = operator_result.data();
+      auto is_correct_type_ptr = is_correct_type.data();
+
+      cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            *is_correct_type_ptr =
+                TestCaseT::template call_operator<DataT, NumElems>(
+                    ref, source_simd_result_data_ptr, operator_result_data_ptr);
+          });
+    });
+    queue.wait_and_throw();
+
+    for (size_t i = 0; i < NumElems; ++i) {
+      {
+        DataT retrieved = source_simd_result[i];
+        DataT expected = ref_data[i];
+        if (!are_bitwise_equal(expected, retrieved)) {
+          passed = false;
+          log::fail(TestDescriptionT(data_type), "Unexpected value at index ",
+                    i, ", retrieved: ", retrieved, ", expected: ", expected);
+        }
+      }
+      {
+        DataT retrieved = operator_result[i];
+        DataT expected = ~shared_ref_data[i];
+        if (!are_bitwise_equal(expected, retrieved)) {
+          passed = false;
+          log::fail(TestDescriptionT(data_type), "Unexpected value at index ",
+                    i, ", retrieved: ", retrieved, ", expected: ", expected);
+        }
+      }
+    }
+
+    if (!is_correct_type.value()) {
+      passed = false;
+      log::print_line("Test failed due to type of the object that returns " +
+                      TestCaseT::get_description() +
+                      " operator is not equal to the expected one for simd<" +
+                      data_type + ", " + std::to_string(NumElems) + ">.");
```

Comment:
```suggestion
      passed = false;
      log::fail(TestDescriptionT(data_type), "Invalid return type for operator");
```
Consider to use `log::fail` for failure logs. The message itself is up to you